main.go

```diff
@@ -10,6 +10,30 @@ import (
 	"strings"
 )
 
+func forwardRequest(r *http.Request, target *url.URL, out chan *http.Response, cancel chan struct{}, fail chan struct{}) {
+	req_a := *r
+	req_a.URL, _ = url.Parse(r.URL.String())
+	req_a.URL.Host = target.Host
+	req_a.Cancel = cancel
+
+	req_a.Close = true
```

Comment:
We might need to close the upstream response body though, once we've finished copying it into the downstream response.

>  The client must close the response body when finished with it: 

-- https://golang.org/pkg/net/http/